bonding-farm/contracts/AutoStakingBond.sol

```diff
@@ -0,0 +1,888 @@
+    mapping(uint256 => uint256) public depositIndexes; // each depositId and its index in ownedDeposits
+    mapping(address => uint256) public depositCounts; // each wallet total deposit count
+
+    mapping(uint256 => Bond) public bondInfo; // stores bond information for depositId
+
+    uint256[] public lockingPeriods; // stores locking periods of discounts
+    mapping(uint256 => uint256) public lockingDiscounts; // stores discount in hundreths for locking periods ( 500 = 5% = 0.05 )
+
+    uint256 public totalRemainingPayout; // total remaining rewardToken payout for bonding
+    uint256 public totalBondedValue; // total amount of payout assets sold to the bonders
+    mapping(address => uint256) public totalPrincipals; // total principal bonded through this depository
+
+    uint256 public minimumPrice; //min price
+
+    string public name; // name of this bond
+
+    string public constant VERSION = '3.0'; // version number
+
+    enum CONFIG {
+        DEPOSIT_TOKEN,
+        FEE_RECIPIENT,
+        FUND_RECIPIENT
+    }
+    mapping(CONFIG => bool) public initialized;
+    uint256 constant ONEinBPS = 10000;
+    uint256 public feeBps; // 10000=100%, 100=1%
+    address public fundRecipient;
+    mapping(address => mapping(address => uint256)) public tokenBalances; // balances for each deposit token
+    // address[] public depositTokens;
+    address[] public feeRecipients;
+    uint256[] public feeWeightBps;
+    mapping(address => uint256) feeWeightFor; // feeRecipient=>feeWeight
+    bool public stakingFee;
+
+    /* ======== STRUCTS ======== */
+
+    // Info for bond holder
+    struct Bond {
+        uint256 depositId; // deposit Id
+        address principal; // token used to create bond
+        uint256 amount; // princial deposited amount
+        uint256 payout; // rewardToken remaining to be paid
+        uint256 vesting; // Blocks left to vest
+        uint256 lastBlockAt; // Last interaction
+        uint256 pricePaid; // In DAI, for front end viewing
+        address depositor; // deposit address
+        bool stake; // staked into the lock farm
+        uint256 fnftId; // lock farm fnft Id
+    }
+
+    /* ======== INITIALIZATION ======== */
+
+    function initialize(
+        string memory _name,
+        address _rewardToken,
+        address _DAO,
+        address _priceOracleAggregator,
+        address _lockFarm,
+        address _fnft,
+        address _tokenVault
+    ) external initializer {
+        require(_rewardToken != address(0));
+        rewardToken = _rewardToken;
+        require(_DAO != address(0));
+        DAO = _DAO;
+        require(_priceOracleAggregator != address(0));
+        priceOracleAggregator = IPriceOracleAggregator(_priceOracleAggregator);
+
+        require(_lockFarm != address(0));
+        lockFarm = ILockFarm(_lockFarm);
+        IERC20Upgradeable(_rewardToken).safeApprove(_lockFarm, 2**256 - 1);
```

Comment:
@bejaxer 
Reentrancy in AutoStakingBond.initialize(string,address,address,address,address,address,address) (bonding-farm/contracts/AutoStakingBond.sol#207-237):
        External calls:
        - IERC20Upgradeable(_rewardToken).safeApprove(_lockFarm,2 ** 256 - 1) (../Backup/TestContract/Masterchef.sol#225)
        - fnft.setApprovalForAll(_tokenVault,true) (../Backup/TestContract/Masterchef.sol#229)
        State variables written after the call(s):
        - __Ownable_init() (../Backup/TestContract/Masterchef.sol#235)
                - _initialized = true (@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol#39)
        - __Pausable_init() (../Backup/TestContract/Masterchef.sol#236)
                - _initialized = true (@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol#39)
        - __Ownable_init() (../Backup/TestContract/Masterchef.sol#235)
                - _initializing = true (@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol#38)
                - _initializing = false (@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol#45)
        - __Pausable_init() (../Backup/TestContract/Masterchef.sol#236)
                - _initializing = true (@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol#38)
                - _initializing = false (@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol#45)
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#reentrancy-vulnerabilities-1